hs_file_types/views.py

```diff
@@ -208,6 +210,91 @@ def remove_aggregation(request, resource_id, hs_file_type, file_type_id, **kwarg
     return JsonResponse(response_data, status=status.HTTP_200_OK)
 
 
+@login_required
+def delete_aggregation(request, resource_id, hs_file_type, file_type_id, **kwargs):
+    """Deletes all files associated with an aggregation and all the associated metadata.
+    """
+
+    response_data = {'status': 'error'}
+
+    res, _, _ = authorize(request, resource_id, needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+    if res.resource_type != "CompositeResource":
+        err_msg = "Aggregation type can be deleted only in composite resource."
+        response_data['message'] = err_msg
+        return JsonResponse(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if hs_file_type not in FILE_TYPE_MAP:
+        err_msg = "Unsupported aggregation type. Supported aggregation types are: {}"
+        err_msg = err_msg.format(FILE_TYPE_MAP.keys())
+        response_data['message'] = err_msg
+        return JsonResponse(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    content_type = ContentType.objects.get(app_label="hs_file_types", model=hs_file_type.lower())
+    logical_file_type_class = content_type.model_class()
+    aggregation = logical_file_type_class.objects.filter(id=file_type_id).first()
+    if aggregation is None:
+        err_msg = "No matching aggregation was found."
+        response_data['message'] = err_msg
+        return JsonResponse(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    res_files = []
+    res_files.extend(aggregation.files.all())
+    aggregation.remove_aggregation()
+    for file in res_files:
+        file.delete()
```

Comment:
wondering why putting aggregation.files.all() into res_files list for iteration is needed here. Wouldn't it work if you directly iterate over aggregation.files.all()?

Reply:
you're right, removed.